shpc/main/modules/base.py

```diff
@@ -338,30 +338,29 @@ def check(self, module_name):
         at updates for entire tags. If a specific folder is provided with
         a container, check the digest.
         """
-        module = self.new_module(module_name)
+        module = self.new_module(module_name, in_registry=True)
         if not os.path.exists(module.module_dir):
             logger.exit(
                 "%s does not exist. Is this a known registry entry?" % module.module_dir
             )
 
         return module.check()
 
-    def new_module(self, name, tag=None, tag_exists=True):
+    def new_module(self, name, tag=None, in_registry=False):
         """
         Create a new module
         """
         name = self.add_namespace(name)
 
-        # If the module has a version, overrides provided tag
-        if ":" in name:
-            name, tag = name.split(":", 1)
-
-        module = Module(name)
-        module.config = self._load_container(module.name, tag)
-
         # Ensure the tag exists, if required, uses config.tag
-        if tag_exists:
-            module.validate_tag_exists()
+        if in_registry:
```

Comment:
For design of `get_module` - we could actually put the `in_registry` param (and checks here) entirely there, then the `get_module` function can be used on its own (with a default for True for `in_registry`. In this spot, we would basically just do:

```python
module = self.get_module(name, tag=tag, in_registry=in_registry)
```
Let me know if that doesn't make sense - it's basically adding this new function and adding some of this "in registry" logic there, since a "get" makes more sense to be checking for this over a function to "create a new one" (which we would assume not to be existing yet!)